packages/firestore/test/unit/specs/recovery_spec.test.ts

```diff
@@ -114,6 +136,31 @@ describeSpec('Persistence Recovery', ['no-ios', 'no-android'], () => {
     }
   );
 
+  specTest(
+    'Ignores intermittent lease refresh failures (with recovery)',
+    ['multi-client'],
+    () => {
+      return client(0)
+        .expectPrimaryState(true)
+        .client(1)
+        .expectPrimaryState(false)
+        .client(0)
+        .failDatabaseTransactions('updateClientMetadataAndTryBecomePrimary')
+        .runTimer(TimerId.ClientMetadataRefresh)
+        .client(1)
+        .failDatabaseTransactions('updateClientMetadataAndTryBecomePrimary')
+        .runTimer(TimerId.ClientMetadataRefresh)
+        .client(0)
+        .recoverDatabase()
+        .runTimer(TimerId.ClientMetadataRefresh)
+        .expectPrimaryState(true)
+        .client(1)
+        .recoverDatabase()
+        .runTimer(TimerId.ClientMetadataRefresh)
```

Comment:
What happens if client 1 happens to run their metadata refresh timer first?

Reply:
I added comment and a second section that verifies the reverse. The primary state doesn't transition during an IndexedDB failure and the order here should not affect which tab is primary.

Note that this is because of: https://github.com/firebase/firebase-js-sdk/blob/master/packages/firestore/src/local/indexeddb_persistence.ts#L433